File: Go/0003.go
package main

import (
    "fmt"
)

// Function to find the largest prime factor
func largestPrimeFactor(n int64) int64 {
    var largest int64 = 0

    // Remove all factors of 2
    for n%2 == 0 {
        largest = 2
        n /= 2
    }

    // Check for odd factors up to sqrt(n)
    for i := int64(3); i*i <= n; i += 2 {
        for n%i == 0 {
            largest = i
            n /= i
        }
    }

    // If remaining n is a prime number > 2
    if n > 2 {
        largest = n
    }

    return largest
}

func main() {
    number := int64(600851475143)
    result := largestPrimeFactor(number)
    fmt.Printf("The largest prime factor of %d is %d\n", number, result)
}

File: Swift/0003.swift
import Foundation

func largestPrimeFactor(of number: Int64) -> Int64 {
    var n = number
    var maxPrime: Int64 = -1

    // Step 1: Remove all factors of 2
    while n % 2 == 0 {
        maxPrime = 2
        n /= 2
    }

    // Step 2: Try odd factors from 3 upwards
    var i: Int64 = 3
    while i * i <= n {
        while n % i == 0 {
            maxPrime = i
            n /= i
        }
        i += 2
    }

    // Step 3: If remaining n is greater than 2, it's a prime
    if n > 2 {
        maxPrime = n
    }

    return maxPrime
}

let number: Int64 = 600851475143
let result = largestPrimeFactor(of: number)
print("The largest prime factor of \(number) is \(result)")